components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt

```diff
@@ -154,6 +172,56 @@ open class GleanInternalAPI internal constructor () {
         return uploadEnabled
     }
 
+    /**
+     * Handles the changing of state when uploadEnabled changes.
```

Comment:
I wonder if we should note here that it should only be called if the flag actually changes.

(I realize this is internal stuff, I'm just thinking about future-us having to deal with it)